### ISSUE TITLE
VectorStore add embedded texts interface

### ISSUE THREAD

**Issue by bhperry** [hwchase17/langchain#5945]
### Feature request

Add a way to pass pre-embedded texts into the VectorStore interface.

Options:
- Standardize the add_embeddings function that has been added to some of the implementations. e.g.
```
    def add_embeddings(
        self,
        texts: List[str],
        embeddings: List[List[float]],
        metadatas: Optional[List[dict]] = None,
        **kwargs: Any,
    ) -> List[str]:
```
- Add embeddings kwarg to the add_texts interface
```
    def add_texts(
        self,
        texts: Iterable[str],
        metadatas: Optional[List[dict]] = None,
        embeddings: Optional[List[List[float]]] = None,
        **kwargs: Any,
    ) -> List[str]:
```

First option is nice in that it leaves each function with a very distinct role, but has added overhead for implementing the interface. Results in very similar code between add_texts and add_embeddings and/or a third private method to handle the actual add operation.

Second option should be pretty straightforward to add to all the implementations, but adds some clutter to the add_texts interface.

### Motivation

Embedding a large body of text (e.g. pubmed) takes a long time, and it is too restrictive to rely on each VectorStore implementation calling embedding functions in the most optimal way every use case. By having a defined way to pass in embeddings directly the interface becomes much more flexible.

For example I've been using huggingface's Datasets.map for processing texts, running embeddings on multiple gpus, etc. Would like to be able to save the final embedded dataset, and then insert into the vector store.

### Your contribution

Happy to help work on it. I know there's quite a few vector store implementations that would have to be updated for this.

**Comment by talhaanwarch** [comment]
In Milvus, these is pretty easy. Look at this page
https://milvus.io/docs/v1.1.1/insert_delete_vector_python.md

**Comment by bhperry** [comment]
Great! I expect most should be straightforward to implement. I would be shocked if there is a vector store that doesn't support this.

**Comment by GomathyDhanya** [comment]
Yeah, decoupling the embedding process and the indexing process would help!

**Comment by bhperry** [comment]
Still relevant
